src/main/java/com/palantir/gradle/externalpublish/ExternalPublishJarPlugin.java

```diff
@@ -38,32 +37,17 @@ public final void apply(Project project) {
     private static void configureJars(Project project) {
         project.getPluginManager().apply(JavaLibraryPlugin.class);
 
+        // The jar manifest may be configured before project versions have been set,
+        // particularly for subprojects which are configured via 'allprojects' or 'subprojects'.
+        Provider<String> versionProvider =
+                project.provider(() -> project.getVersion().toString());
+
         project.getTasks().withType(Jar.class).named("jar").configure(jar -> {
-            jar.getManifest()
-                    .attributes(
-                            Collections.singletonMap("Implementation-Version", new ProjectVersionToString(project)));
+            jar.getManifest().attributes(Collections.singletonMap("Implementation-Version", versionProvider));
```

Comment:
I suspect this is still going to hold onto project; but worth testing locally.

Reply:
Good call, I've tested it. On gradle-conjure, `./gradlew classes --configuration-cache` works, and the Implementation-Version is correctly set in the manifest.

Reply:
From in-person conversation: I am so glad this actually seems to work! It looks like the javadoc is wrong, Gradle must have got this to work at some point.

Reply:
I'm very curious how they determine that providers mapped from `Project` don't get serialised

Reply:
Yeah, I'm a bit stumped about that too. I guess there's some magic involved.